src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -656,39 +866,83 @@ public void setHeartbeatFrequency(int frequency) {
     }
   }
 
-  /** Retrieve session heartbeat frequency in seconds */
+  /**
+   * Retrieve session heartbeat frequency in seconds
+   *
+   * @return the heartbeat frequency in seconds
+   */
   public int getHeartbeatFrequency() {
     return this.heartbeatFrequency;
   }
 
+  /**
+   * Get autocommit
```

Comment:
In general I don't really see the point of such doc strings since there's no value added on top of what the code provides by itself. It'd be much more useful if it explained what the autocommit field means. On setters we could link the definitions provided for getters e.g.

```java
  /**
   * autoCommit field specifies whether autocommit is enabled for the session.
   * Autocommit determines whether a DML statement, when executed without an active transaction,
   * is automatically committed after the statement successfully completes. For more information,
   * default: true
   *
   * @see <a href="https://docs.snowflake.com/en/sql-reference/transactions#label-txn-autocommit">Transactions/Autocommit</a>
   *
   * @return a boolean value of autocommit field
   */
  public boolean getAutoCommit() {
    return autoCommit.get();
  }

  /**
   * Sets value of autoCommit field
   * @see SFBaseSession#getAutoCommit()
   *
   * @param autoCommit - boolean
   */
  public void setAutoCommit(boolean autoCommit) {
    this.autoCommit.set(autoCommit);
  }
```